### PR TITLE
Docs: fix broken links from recent Algolia crawl

### PR DESCRIPTION
**Pull request by dangerismycat**
I noticed a ton of broken/old links from our recent crawl (301/302 and 500 responses). This PR updates those links.

### CODE REVIEW

File: docs/markdown/testing/redbox.md [hunk 1]
```diff
@@ -210,18 +210,18 @@ Some interactive example to get the best out of redbox
 
 ## Resources
 
-**[RedboxGroup](/buttongroup)**
+**[RedboxGroup](/web/buttongroup)**
```
Comment (ayeshakmaz) [inline docs/markdown/testing/redbox.md:213]: Should we keep these testing files around? Seems like more work to keep them updated

Comment (dangerismycat) [reply docs/markdown/testing/redbox.md:213]: I'd be fine with deleting them. I think @rlingineni wanted to keep them around for reference but that's what git history is for 🙂

I'll put up another PR to remove them

Comment (dangerismycat) [reply docs/markdown/testing/redbox.md:213]: here's that PR: https://github.com/pinterest/gestalt/pull/2378

File: playwright/accessibility/Typography.spec.mjs [hunk 2]
```diff
@@ -3,6 +3,6 @@ import { test } from '@playwright/test';
 import expectAccessiblePage from './expectAccessiblePage.mjs';
 
 test('Typography Accessibility check', async ({ page }) => {
-  await page.goto('/web/typography');
```
Comment (ayeshakmaz) [inline playwright/accessibility/Typography.spec.mjs:6]: Shouldn't this fail? surprised it didn't complain

Comment (dangerismycat) [reply playwright/accessibility/Typography.spec.mjs:6]: Gooooood question

File: docs/markdown/testing/redbox.md (hunk 1 above)
Comment (ayeshakmaz) [reply docs/markdown/testing/redbox.md:213]: Thank you!

**Pull request merged**